compiler/noirc_evaluator/src/brillig/brillig_gen/brillig_block.rs

```diff
@@ -1705,6 +1690,115 @@ impl<'block> BrilligBlock<'block> {
         }
     }
 
+    fn initialize_constant_array(
+        &mut self,
+        data: &im::Vector<ValueId>,
+        typ: &Type,
+        dfg: &DataFlowGraph,
+        pointer: MemoryAddress,
+    ) {
+        if data.is_empty() {
+            return;
+        }
+        let item_types: &Vec<_> = match typ {
+            Type::Array(item_types, _) => item_types.borrow(),
+            Type::Slice(item_types) => item_types.borrow(),
+            _ => unreachable!("ICE: Array constant is not of array or slice type {typ:?}"),
+        };
+
+        // Find out if we are repeating the same item over and over
+        let mut data_as_unique_items = HashSet::default();
+        for item_index in (0..data.len()).step_by(item_types.len()) {
+            let item: Vec<_> = (0..item_types.len()).map(|i| data[item_index + i]).collect();
+            data_as_unique_items.insert(item);
+        }
+
+        // If all the items are single address, and all have the same initial value, we can initialize the array in a runtime loop.
+        // Since the cost in instructions for a runtime loop is in the order of magnitude of 10, we only do this if the item_count is bigger than that.
+        let item_count = data.len() / item_types.len();
+
+        if item_count > 10
+            && data_as_unique_items.len() == 1
+            && item_types.iter().all(|typ| matches!(typ, Type::Numeric(_)))
+        {
+            let mut subitem_to_repeat_variables = Vec::with_capacity(item_types.len());
+            for subitem_id in data_as_unique_items.into_iter().next().unwrap().into_iter() {
+                subitem_to_repeat_variables.push(self.convert_ssa_value(subitem_id, dfg));
+            }
+
+            let data_length_variable =
+                self.brillig_context.make_usize_constant_instruction(data.len().into());
+
+            // If this is an array with complex subitems, we need a custom step in the loop to write all the subitems while iterating.
+            if item_types.len() > 1 {
+                let step_variable =
+                    self.brillig_context.make_usize_constant_instruction(item_types.len().into());
+
+                let subitem_pointer =
+                    SingleAddrVariable::new_usize(self.brillig_context.allocate_register());
+
+                self.brillig_context.codegen_loop_with_bound_and_step(
+                    data_length_variable.address,
+                    step_variable.address,
+                    |ctx, iterator_register| {
```

Comment:
Can you try breaking out some of these variables onto multiple lines to improve the formatting and reduce the overall line count?

Reply:
Extracted `FnOnce`s to variables which helped a bit, but extracting `_variable.address` to their own variables didn't change formatting